Don't flip catcher plate contents when catcher changes direction

ekrctb opened the pull request:
- Fixes #11112

I added a skin configuration key for whether the catcher plate should be flipped.
It is `true` for the default skin, and `false` for any legacy skin. Not sure how to make it customizable any further.

ekrctb committed:
Decouple direction of catcher from its scale

ekrctb committed:
Don't flip catcher plate contents in legacy skin

ekrctb committed:
Add test for catcher plate flipping configuration

bdach commented on osu.Game.Rulesets.Catch/Skinning/Legacy/CatchLegacySkinTransformer.cs:
```diff
@@ -103,6 +103,16 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
+                case CatchSkinConfiguration config:
+                    switch (config)
+                    {
+                        case CatchSkinConfiguration.FlipCatcherPlate:
+                            // Always return `false` (don't flip catcher plate contents) for a legacy skin.
+                            return (IBindable<TValue>)new Bindable<bool>();
```
The way this is currently done means that the flipping will not take place on the default skin either if beatmap skins are turned on.

Unsure how to resolve this but seems like unintentional behaviour.

frenzibyte replied:
It may seem like an unintentional behaviour, but the eventual plan for legacy beatmap skins is to be backed by a default legacy skin if the user skin is not legacy, so it makes some sort of sense for this to remain here, I think.

peppy replied:
To fix that, this should probably have a pre-check on whether a catcher is provided, and return null if it isn't.

```csharp
if (this.GetDrawableComponent(Catcher) == null)
    return null;
```

ekrctb committed:
Fix default skin catcher not flipping catcher plate

When legacy beatmap skin is present but catcher is not provided, it was using the legacy setting (always false).

peppy commented on osu.Game.Rulesets.Catch/Skinning/CatchSkinConfiguration.cs:
```diff
@@ -0,0 +1,13 @@
+        /// <summary>
+        /// Whether the contents of the catcher plate should be visually flipped when the catcher direction is changed.
+        /// </summary>
+        FlipCatcherPlate
```
Still not sure about adding new values to these areas (was intended to only be for actually supported values in osu-stable skin versions) but I don't have a better suggestion for now.
